### PR TITLE
fix: add hook display_login in error login page

### PR DESCRIPTION
**Pull request by MyvTsv**
- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.

## Description

- It fixes !35394
Added a display_login hook in login_error.html.twig which allows to change the logo thanks to the branding plugin

## Screenshots (if appropriate):

Capture d’écran du 2024-11-29 08-52-09

### CODE REVIEW

**Review by stonebuzz — APPROVED** [review]

File: templates/pages/login_error.html.twig [hunk 1]
```diff
@@ -53,6 +53,8 @@
                 <i class="ti ti-login"></i>
                 <span>{{ __('Log in again') }}</span>
             </a>
+
+            {{ call_plugin_hook(constant('Glpi\\Plugin\\Hooks::DISPLAY_LOGIN')) }}
```
Comment (stonebuzz) [inline templates/pages/login_error.html.twig:57]: Maybe a new hook `DISPLAY_LOGIN_ERROR` should be created to differentiate between the login page and the login error page 

If I take the example of the news plugin (which lets you add information to the home page), this change will also display the information on the error page

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
This will not fix some specific pages, like the password recovery pages. The best solution,  IMHO, is to add 3 new hooks, `add_css_notlogged`, `add_javascript_notlogged` and `add_header_tag_notlogged`, and use them in the `/templates/layout/page_card_notlogged.html.twig` context. This will provide a generic and flexible solution that could be used more widely than hooks dedicated to precise pages.

**Review by cedric-anne — CHANGES_REQUESTED** [review]
1. The logic could be moved in the `FrontEndAssetsExtension`. Indeed, you could declare Twig function like `get_plugins_js_files(bool $is_anonymous_page)` that would do the same as your `Html::add_css_file_plugin()` method. Then it could be used in the `templates/layout/parts/head.html.twig` and the `templates/layout/parts/page_footer.html.twig` files like this: `{% set js_file = js_file|merge(get_plugins_js_files(is_anonymous_page|default(false))) %}`. You would just then have to remove this logic from the `Html::footer()` and from the `Html::includeHeader()` methods.

2. Maybe replacing the `_notlogged` suffix by `_anonymous` or even `_anonymous_page` would be less ambiguous for plugin developers.

3. The same logic should also be applied to the `add_javascript_module` hook. I did not see it previously because the corresponding constant is not declared. The constant should be added in the `Hooks` class.

**Review by cedric-anne — APPROVED** [review]
@MyvTsv I made a few changes, you can check them commit by commit.

I tested it manually and it is OK for me. However, I would prefer to have a feedback from the customer before merging this.

Also, could you please add an example usage for all these new hooks in the [example plugin](https://github.com/pluginsGLPI/example) ?

**Review by stonebuzz — APPROVED** [review]
Ok for me, 

You also need to add the new Hook `ADD_*_ANONYMOUS_*` to the GLPI doc.

https://github.com/glpi-project/docdev

**Review by Rom1-B — APPROVED** [review]

**Review by trasher — APPROVED** [review]
OK for me, still need to be documented.

**Comment by stonebuzz** [comment]
> OK for me, still need to be documented.

@MyvTsv -> here -> https://github.com/glpi-project/doc (with new `PR`)

**Comment by trasher** [comment]
> > OK for me, still need to be documented.
> 
> @MyvTsv -> here -> https://github.com/glpi-project/doc (with new `PR`)

Open against master branch since it's a 10.0 change.

**Pull request merged**